container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go

```diff
@@ -1786,46 +1794,28 @@ func newContainersListFromDockerContainersList(dockerContainers []types.Containe
 	return containers, nil
 }
 
-func newContainerFromDockerContainer(dockerContainer types.Container) (*docker_manager_types.Container, error) {
-	containerStatus, err := getContainerStatusByDockerContainerState(dockerContainer.State)
-	if err != nil {
-		return nil, stacktrace.Propagate(err, "An error occurred getting ContainerStatus from Docker container state '%v'", dockerContainer.State)
-	}
-	containerName, err := getContainerNameByDockerContainerNames(dockerContainer.Names)
+func newContainerFromDockerContainer(dockerContainer types.ContainerJSON) (*docker_manager_types.Container, error) {
+	containerStatus, err := getContainerStatusByDockerContainerState(dockerContainer.State.Status)
 	if err != nil {
-		return nil, stacktrace.Propagate(err, "An error occurred getting container name from Docker container names '%+v'", dockerContainer.Names)
+		return nil, stacktrace.Propagate(err, "An error occurred getting ContainerStatus from Docker container state '%v'", dockerContainer.State.Status)
 	}
-
-	// Frustratingly, Docker's ContainerList returns ports in a completely different format than ContainerInspect, so we need
```

Comment:
`containerJSON` uses the required format for ports so we don't need that extra logic anymore.